typescript.js

```diff
@@ -9,15 +9,19 @@ module.exports = {
       parserOptions: { warnOnUnsupportedTypeScriptVersion: false },
       extends: ['plugin:@typescript-eslint/recommended'],
       rules: {
-        'no-use-before-define': 'off', // Turning off because we have the TS version off
-        '@typescript-eslint/no-use-before-define': 'off',
-        '@typescript-eslint/explicit-function-return-type': ['error', { allowExpressions: true }],
-
+        'no-use-before-define': 'off', // @typescript-eslint/no-use-before-defined requires this to be off
+        '@typescript-eslint/no-use-before-define': ['error', { functions: false }],
         'no-unused-vars': 'off', // @typescript-eslint/no-unused-vars requires this to be off
-        '@typescript-eslint/no-unused-vars': ['error', { argsIgnorePattern: '^_' }],
-
-        // TS complains about using the .ts/.tsx extensions in the import statement. so turning this rule off for TS
-        'import/extensions': 'off',
+        'no-empty-functions': 'off', // @typescript-eslint/no-empty-function requires this to be off
+        '@typescript-eslint/no-empty-function': ['error', { allow: ['arrowFunctions'] }],
+        '@typescript-eslint/consistent-type-imports': 'error',
```

Comment:
Note from Jordi:
@typescript-eslint/consistent-type-imports -- error instead of warn; this separates type and non-type imports and it is automatically fixable. https://typescript-eslint.io/rules/consistent-type-imports/ (For example makes this change automatically:

![image](https://user-images.githubusercontent.com/3885959/215878357-8cc06f88-2683-4a99-bc20-0d720e50744e.png)

It is nice because it is easy to tell in your imports which ones are types and which ones are not, and it is sorted automatically for you when you save if you do eslint fixes on save.

Doesn't turn off import/extensions for TS files because it should work properly now.